### PR TITLE
Disallow inline array operations in expression trees.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Relates to test plan https://github.com/dotnet/roslyn/issues/67826

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs [hunk 1]
```diff
@@ -4241,6 +4263,36 @@ static async Task<T> FromResult<T>(T r)
...
+                //         () => x[0];
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsInlineArrayOperation, "x[0]").WithLocation(7, 15),
+                // (10,15): error CS0832: An expression tree may not contain an assignment operator
+                //         () => x[0] = 111;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsAssignment, "x[0] = 111").WithLocation(10, 15),
+                // (10,15): error CS9507: An expression tree may not contain an inline array access or conversion
+                //         () => x[0] = 111;
+                Diagnostic(ErrorCode.ERR_ExpressionTreeContainsInlineArrayOperation, "x[0]").WithLocation(10, 15)
+                );
+        }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/InlineArrayTests.cs:4294]: Are we testing the conversion case? "An expression tree may not contain an inline array [...] conversion"

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by AlekseyTs** [comment]
@cston, @jjonescz, @dotnet/roslyn-compiler For the second review.

**Review by jjonescz — APPROVED** [review]

**Pull request merged**